### ISSUE TITLE
x64 programs are not automatically rendered with MacType in 2018.10.19 beta

### ISSUE THREAD

**Issue by eroneko** [snowie2000/mactype#478]
I clean installed mactype 10.19 beta and selected registry mode but could ony get x86 programs to be automatically rendered with mactype. Manually enabling x64 programs with mactype in process manager is working.
I'm using Windows 10 1809 LTSC.
![qq 20181123203018](https://user-images.githubusercontent.com/5058776/48943627-b9ebc200-ef5e-11e8-9349-70e6fceec696.png)

**Comment by snowie2000** [comment]
You should check your registry for correction, which is kind of complicated.

Try this easy solution: Start->Run->MacType64.dll, it should give you a warning of opening a DLL file or ask you to select a program to open it. If it cannot find the file, there is something wrong with your environment variable settings.

Then you should check what in the registry position described here: https://support.microsoft.com/en-gb/help/197571/working-with-the-appinit-dlls-registry-value
Post what that is to see if we can help.

**Comment by eroneko** [comment]
> You should check your registry for correction, which is kind of complicated.
> 
> Try this easy solution: Start->Run->MacType64.dll, it should give you a warning of opening a DLL file or ask you to select a program to open it. If it cannot find the file, there is something wrong with your environment variable settings.
> 
> Then you should check what in the registry position described here: https://support.microsoft.com/en-gb/help/197571/working-with-the-appinit-dlls-registry-value
> Post what that is to see if we can help.

For the first solution, it does give me a warning of opening a dll file so maybe it has nothing to do with my environment variable settings.
These are what I have in HKEY_LOCAL_MACHINE\Software\Microsoft\Windows NT\CurrentVersion\Windows 
![qq 20181123212922](https://user-images.githubusercontent.com/5058776/48945913-eefc1280-ef66-11e8-9768-a889ebd87e6a.png)

**Comment by snowie2000** [comment]
Appinit_dlls must have a "mactype64.dll" in it. It was probably denied by your security software. 

Write to it manually should fix the problem.

**Comment by sammilucia** [comment]
I had this problem when I first installed MacType .. MacWiz said it was already in Registry Mode, however AppInit_DLLs wasn't set .. And clicking Next Next Finish in MacWiz (in Registry Mode) didn't add it.

**Comment by snowie2000** [comment]
That's why your x86 applications are working correctly.

**Comment by snowie2000** [comment]
> I had this problem when I first installed MacType .. MacWiz said it was already in Registry Mode, however AppInit_DLLs wasn't set .. And clicking Next Next Finish in MacWiz (in Registry Mode) didn't add it.

That's because the wizard checks the x86 part of the registry. If it successfully found the expected key, it'll treat the start mode as registry.

When you didn't change the loading mode (at least the wizard thinks so), it does nothing.

**Comment by sammilucia** [comment]
Yes I know ... I just suspect there's a set of conditions where MacWiz thinks it's in Registry Mode but MacType isn't actually working.

**Comment by sammilucia** [comment]
@snowie2000 Maybe I should clear both registry areas when installing the latest version? ... Yes, I need to add this function.

**Comment by sammilucia** [comment]
@snowie2000 MacWiz should also be checking *both* registry strings and making sure they're consistent.

**Comment by sammilucia** [comment]
@snowie2000 I assume it will only be MacType.dll (not MacType64.dll) that appears in HKEY_LOCAL_MACHINE\SOFTWARE\Wow6432Node\Microsoft\Windows NT\CurrentVersion\Windows
?

**Comment by snowie2000** [comment]
Yes, wow6432node only needs the mactype.dll file

I'll complete the check.

**Comment by sammilucia** [comment]
@snowie2000 yep confirmed. Changes to the installer are done, I'll commit them shortly

**Comment by sammilucia** [comment]
@snowie2000 should be pushed. As you know I've been having SSL problems which should now be fixed. Can you please confirm you can see that commit on Inno-setup branch?

You can merge the Inno-setup branch if you want

**Comment by snowie2000** [comment]
>Samantha Glocker | 571ced09bc Added clean-up of 32-bit AppInit_DLLs | 5 hours ago
>Samantha Glocker | 183f2429f9 Icons added | 4 days ago

Confirmed.

**Comment by sammilucia** [comment]
Great thank you.

**Comment by fonlan** [comment]
Pick a small stool to wait for the new beta^_^

**Comment by sergeevabc** [comment]
New beta is released. Does it resolve the complication?

**Comment by Ray-Eldath** [comment]
No more update on this project...?
